src/platform/plugins/shared/chart_expressions/expression_partition_vis/public/utils/layers/get_color.test.ts

```diff
@@ -19,371 +19,390 @@ import { ChartTypes } from '../../../common/types';
 import { getDistinctSeries } from '..';
 import { getColorCategories } from '@kbn/chart-expressions-common';
 
-describe('#byDataColorPaletteMap', () => {
-  let paletteDefinition: PaletteDefinition;
-  let palette: PaletteOutput;
-  let colorIndexMap: Map<string, number>;
-  const visData = createMockVisData();
-  const categories = (chartType?: ChartTypes) =>
-    chartType === ChartTypes.MOSAIC && visData.columns.length === 2
-      ? getColorCategories(visData.rows, visData.columns[1]?.id)
-      : getColorCategories(visData.rows, visData.columns[0]?.id);
+describe('get color', () => {
+  describe('#byDataColorPaletteMap', () => {
+    let paletteDefinition: PaletteDefinition;
+    let palette: PaletteOutput;
+    let colorIndexMap: Map<string, number>;
+    const visData = createMockVisData();
+    const categories = getColorCategories(visData.rows, visData.columns[1]?.id);
```

Comment:
This method is only called from the when there is a second layer on Mosaic so no need to conditionalize the categories here.